src/tmpfiles/tmpfiles.c

```diff
@@ -2260,6 +2283,69 @@ static int remove_item(Item *i) {
         }
 }
 
+static int format_age_by(const Item *item, char **ret_age_by_fmt) {
```

Comment:
this function is only used during debug logging, and is awfully complex for that, and generates a format that is quite distinct from the format we parse. why not something like this:

```c
static char *age_by_to_string(AgeBy a, bool uppercase) {
        static const char time_map[] = { 'A', 'B', 'C', 'M' };
        char *ret;
        size_t j = 0;

        ret = new(char, ELEMENTSOF(time_map) + 1);
        if (!ret)
                return NULL;

        for (size_t i = 0; i < ELEMENTSOF(time_map); i++)
                if (FLAGS_SET(a, 1U << i))
                        ret[j++] = time_map[i] + (uppercase ? 0 : 'a' - 'A');

        *ret = 0;
        return 0;
}
```

and then invoke that twice, once for the dirs and once for the files?

Reply:
Yeah, your version looks simpler. Will replace that function.